src/rules/dto-decorators.ts

```diff
@@ -0,0 +1,221 @@
+import { Rule, SourceCode } from "eslint";
+import {
+  ClassDeclaration,
+  Directive,
+  ModuleDeclaration,
+  Program,
+  Statement,
+  Node,
+  ExportDefaultDeclaration,
+} from "estree";
+import { findImportsByPackageName } from "../utils/imports-finder";
+
+interface ProficoNodeBodyElement {
+  decorators?: ProficoDecorator[];
+  type: "PropertyDefinition";
+}
+
+interface ProficoDecorator {
+  expression: {
+    callee: {
+      name: string;
+    };
+  };
+  type: "Decorator";
+  range: number[];
+}
+
+const dtoDecorators: Rule.RuleModule = {
+  meta: {
+    fixable: "code",
+    type: "problem",
+    messages: {
+      improperlyOrderedDecorators: "Decorators are not properly ordered.",
+    },
+    docs: {
+      url: "https://github.com/profico/eslint-plugin-profico#dto-decorators",
+    },
+  },
+  create: context => {
+    return {
+      Program({ body }: Program) {
+        const optionalDecorator = new Set(["IsOptional"]);
+        const allNonCustomDecorators = new Set(["IsOptional"]);
+
+        const swaggerDecorators: Set<string> = findImportsByPackageName(
+          body,
+          "@nestjs/swagger",
+        );
+
+        Array.from(swaggerDecorators).forEach(decorator => {
+          allNonCustomDecorators.add(decorator);
+        });
+
+        const classTransformerDecorators: Set<string> =
+          findImportsByPackageName(body, "class-transformer");
+
+        Array.from(classTransformerDecorators).forEach(decorator => {
+          allNonCustomDecorators.add(decorator);
+        });
+
+        const classValidatorDecorators: Set<string> = findImportsByPackageName(
+          body,
+          "class-validator",
+        );
+
+        Array.from(classValidatorDecorators).forEach(decorator => {
+          allNonCustomDecorators.add(decorator);
+        });
+
+        function getOrderedDecorators(
+          decorators: ProficoDecorator[],
+          sourceCode: SourceCode,
+        ) {
+          const swaggerDecs = decorators.filter(dec =>
+            swaggerDecorators.has(dec.expression.callee.name),
+          );
+
+          const classTransformerDecs = decorators.filter(dec =>
+            classTransformerDecorators.has(dec.expression.callee.name),
+          );
+
+          const classValidatorDecs = decorators
+            .filter(
+              dec =>
+                classValidatorDecorators.has(dec.expression.callee.name) &&
+                dec.expression.callee.name !== "IsOptional",
+            )
+            .sort((a, b) => {
+              if (b.expression.callee.name > a.expression.callee.name) {
+                return -1;
+              } else return 1;
+            });
```

Comment:
Can be shortened to `.sort((a, b) => b.expression.callee.name > a.expression.callee.name ? -1 : 1)`